features/facebookImport/src/views/import/import.jsx

```diff
@@ -1,42 +1,118 @@
-import React, { useContext } from "react";
+import React, { useContext, useEffect, useState } from "react";
 import { ImporterContext } from "../../context/importer-context.jsx";
-import Loading from "../../components/loading/loading.jsx";
+import {
+    FileSelectionError,
+    FileImportError,
+} from "../../errors/polyIn-errors.js";
 
 import ProgressBarComponent from "../../components/progressBar/progressBar.jsx";
 import ImportExplanationExpandable from "../../components/importExplanationExpandable/importExplanationExpandable.jsx";
 
 import "./import.css";
 import i18n from "../../i18n.js";
 import PolypolyDialog from "../../components/dialogs/polypolyDialog/polypolyDialog.jsx";
+import { FileLoaderContext } from "../../context/file-loader-context.jsx";
 
 //These are just the sections that are shown as a visual part of the import
 //importSteps are all steps like loading and finished that have logical relevance for the process
 const importSections = ["request", "download", "import", "explore"];
 
+const importSteps = {
```

Comment:
This is actually a kinda-enum. It might be simplified to:

```js
const importSteps = new Map( ["beginning", "request", "download","import","explore"].map( i => [i, i] ));
```

That way, you don't have to repeat the strings twice.

Reply:
`importSteps = Object.fromEntries(new Map( ["beginning", "request", "download","import","explore"].map( i => [i, i] )));` to get the Object right? Is this really better then?

Reply:
Ah, right, you would have to use 

```
const importSteps = Object.fromEntries(new Map( ["beginning", "request", "download","import","explore"].map( i => [i, i] )));
```
but you suggested that somewhere else?

Reply:
Right it's in a previous version of the comment (or whatever). I guess it's marginally better. At the end of the day, we're trying to define an enum here... This is the closest functionally. We could create our own silly-enum class and put it somewhere, because this is probably a pattern that's repeated elsewhere.

Reply:
Yeah was thinking about that, but decided against it as it just seems somewhat overkill

Reply:
Right, there are several instances where our own enum would make sense. Anywhere with a status, for instance. I've mentioned it elsewhere, for instance [here](https://github.com/polypoly-eu/polyPod/pull/435#discussion_r743145997)

Reply:
> Yeah was thinking about that, but decided against it as it just seems somewhat overkill

Alas, I don't know. Not having enums in a modern language certainly seems underkill. For me, overkill would have been using [`Symbol`s](https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Symbol).

But certainly using that every time we want a seudo-enum would be overkill. I've checked out the [enum](https://www.npmjs.com/package/enum) package and it seems overkill too. So maybe just our own little polyEnum would be a nice thing to have, and could find uses anywhere.

So what do you say if I create a jira ticket for that and leave this as it is now? (And thanks for the feedback)

Reply:
Yeah I think thats a very nice idea :)

Reply:
In typescript they are supported, just not in js

Reply:
Done https://jira.polypoly.eu/browse/PROD4POD-1485